broker/aws/impl.go

```diff
@@ -283,30 +283,40 @@ func (b *Broker) withProfileAssumeRole(accountName string, profileName string,
 	return assumeRoleOutput, region, err
 }
 
+const getAWSRolesTimeout = 10 * time.Second
+
 func (b *Broker) withSessionGetAWSRoleList(validSession *session.Session, accountName string) ([]string, error) {
 	iamClient := iam.New(validSession)
 	var maxItems int64
 	maxItems = 500
 	listRolesInput := iam.ListRolesInput{MaxItems: &maxItems}
-
 	var roleNames []string
-	awsListRolesAttempt.WithLabelValues(accountName).Inc()
-	err := iamClient.ListRolesPages(&listRolesInput,
-		func(listRolesOutput *iam.ListRolesOutput, lastPage bool) bool {
-			b.logger.Debugf(2, "listrolesoutput =%v", listRolesOutput)
-			b.logger.Debugf(2, "loop, lastPage=%s", lastPage)
-			for _, role := range listRolesOutput.Roles {
-				roleNames = append(roleNames, *role.RoleName)
-			}
-			return !lastPage
-		})
-	if err != nil {
-		return nil, err
+	c := make(chan error)
```

Comment:
Please make the capacity equal to the number of accounts to iterate over.

Reply:
In this case, capacity should be 1.
As a general rule, channels should not be blocking (capacity=0) unless there is a good reason.

Reply:
thank you fixes